kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -287,6 +280,7 @@ class SparseMatrixMultiplicationUtility
 
         IndexType max_row_width = 0;
 
+        // TODO: Replace with block_for_each. Difficult due to critical section
```

Comment:
you can use [`MaxReduction`](https://github.com/KratosMultiphysics/Kratos/blob/3aacd9741d3ec20c68a5700d43ebdda0be2ad20e/kratos/utilities/reduction_utilities.h#L126-L152) with an [overload of `IndexPartition::for_each`](https://github.com/KratosMultiphysics/Kratos/blob/3aacd9741d3ec20c68a5700d43ebdda0be2ad20e/kratos/utilities/parallel_utilities.h#L194-L214) that takes a reduction.

Reply:
LEts see